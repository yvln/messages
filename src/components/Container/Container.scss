@import "styles/_variables.scss";

.Container {
  background-color: white;
  border: 1px solid $color-grey-light;
  max-width: 700px;
  margin: 0 auto;
}