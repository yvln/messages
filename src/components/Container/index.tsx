import * as React from 'react';

import './Container.scss';

export interface IProps {
  children: React.ReactNode;
  className?: string;
}

const Container = (props: IProps) => {
  return (
    <div className={`Container ${props.className || ''}`}>{props.children}</div>
  );
};

export default Container;
