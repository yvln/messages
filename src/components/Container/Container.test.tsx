import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Container, { IProps } from '.';

configure({ adapter: new Adapter() });

describe('src/components/Container', () => {
  let props: IProps;

  beforeEach(() => {
    props = {
      children: <div>Mama!</div>,
      className: 'Specific-container',
    };
  });

  it('should render', () => {
    const component = shallow(<Container {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should display its children', () => {
    const component = shallow(<Container {...props} />);
    expect(component.text()).toEqual('Mama!');
  });

  it('should have a new className', () => {
    const component = shallow(<Container {...props} />);
    expect(component.hasClass('Container'));
    expect(component.hasClass('Specific-container'));
  });
});
