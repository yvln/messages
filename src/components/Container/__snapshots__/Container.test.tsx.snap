// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Container should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    className="Specific-container"
  >
    <div>
      Mama!
    </div>
  </Container>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Mama!
      </div>,
      "className": "Container Specific-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Mama!",
      },
      "ref": null,
      "rendered": "Mama!",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Mama!
        </div>,
        "className": "Container Specific-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mama!",
        },
        "ref": null,
        "rendered": "Mama!",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
